service/frontend/workflowHandler.go

```diff
@@ -835,6 +835,7 @@ func (wh *WorkflowHandler) PollWorkflowTaskQueue(ctx context.Context, request *w
 		return err
 	}
 
+	nilResponse := &workflowservice.PollWorkflowTaskQueueResponse{}
```

Comment:
I wouldn't create variables likes this. Mainly because it doesn't simplify code (seems to me it makes it even harder to read), and secondly, it is memory allocation which always happen but variable itself is needed in two rare error cases.